File: app/src/main/java/com/imangazalievm/bubbble/domain/interactors/ShotsInteractor.java
package com.imangazalievm.bubbble.domain.interactors;


import com.imangazalievm.bubbble.domain.models.Shot;
import com.imangazalievm.bubbble.domain.models.ShotsRequestParams;
import com.imangazalievm.bubbble.domain.repository.ShotsRepository;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Single;

public class ShotsInteractor {

    private ShotsRepository shotsRepository;

    @Inject
    public ShotsInteractor(ShotsRepository shotsRepository) {
        this.shotsRepository = shotsRepository;
    }

    public Single<List<Shot>> getShots(ShotsRequestParams shotsRequestParams) {
        return shotsRepository.getShots(shotsRequestParams);
    }

}

File: app/src/main/java/com/bubbble/domain/shotslist/ShotsInteractor.kt
package com.bubbble.domain.shotslist

import com.bubbble.data.shots.ShotsRepository
import com.bubbble.core.models.Shot
import com.bubbble.core.models.ShotsRequestParams
import javax.inject.Inject

class ShotsInteractor @Inject constructor(
    private val shotsRepository: ShotsRepository
) {

    suspend fun getShots(shotsRequestParams: com.bubbble.core.models.ShotsRequestParams): List<com.bubbble.core.models.Shot> {
        return shotsRepository.getShots(shotsRequestParams)
    }

}